### PR TITLE
Implement SASS_PATH

### LINKED ISSUE #1678: Is there any plan to support $SASS_PATH?
I know `node-sass` has the `--include-path` option. However, the original Sass documentation mentioned the ability to provide include paths with a `SASS_PATH` environment variable.

This would be very helpful for me, as I need `node-sass` to look in a specific set of include paths *by default* in my environment. Specifying `--include-path` every time is not an option for me in this case.

There was a discussion about [including this in libsass](https://github.com/sass/libsass/issues/90), but they concluded that it should be the wrappers' job, rather than the job of libsass itself.

### PR DESCRIPTION
**Pull request by nottrobin**
This is intended to [implement the `SASS_PATH` environment variable](https://github.com/sass/libsass/issues/90#issuecomment-57822790).

Fixes #1678.

## Tests

I've added two new API tests under `.render`:

- "should check SASS_PATH in the specified order"
- "should prefer include path over SASS_PATH"

If you run `mocha test/api.js` you should see these tests pass.

## Manual checking

You can test it manually using the test fixtures as follows:

### SASS_PATH is picked up

``` bash
$ fixdir=`pwd`/test/fixtures/sass-path
$ export SASS_PATH=$fixdir/red
$ bin/node-sass $fixdir/index.scss
body {
  background: red; }
```

### Earlier paths are preferred

``` bash
$ export SASS_PATH=$fixdir/orange:$fixdir/red
$ bin/node-sass $fixdir/index.scss
body {
  background: orange; }
```

### Specified include-paths still take precedence

``` bash
$ bin/node-sass $fixdir/index.scss --include-path $fixdir/red
body {
  background: red; }
```

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -265,6 +264,18 @@ function normalizeFunctionSignature(signature, callback) {
  */
 
 module.exports.render = function(opts, cb) {
+  // Add include paths from SASS_PATH environment variable, if present
```
Comment (nschonni) [inline lib/index.js]: This normalization probably belongs up in the getOptions `options.includePaths = (options.includePaths || []).join(path.delimiter);`, then there is no change anything else

File: test/api.js [hunk 2]
```diff
@@ -137,6 +137,41 @@ describe('api', function() {
       });
     });
 
+    it('should check SASS_PATH in the specified order', function(done) {
+      var src = read(fixture('include-path/index.scss'), 'utf8');
+      var expected = read(fixture('include-path/expected.css'), 'utf8').trim();
+
+      var separator = process.platform == 'win32' ? ';' : ':';
+      process.env.SASS_PATH = fixture('include-path/functions');
```
Comment (nschonni) [inline test/api.js]: Could be wrong with the Mocha lifecycle for the `env`, but it's probably good to unset the `SASS_PATH` value in an after test function

File: test/fixtures/include-path/lib-alternate/vars.scss [hunk 3]
```diff
@@ -0,0 +1 @@
+$color: orange;
```
Comment (nschonni) [inline test/fixtures/include-path/lib-alternate/vars.scss]: `lib-alternate` might be hard to follow, what about `just keeping it as `SASS_PATH`?

Comment (nottrobin) [inline test/fixtures/include-path/lib-alternate/vars.scss]: Sorry I didn't quite follow that. Do you mean put my fixtures in their own directory called "sass-path" or something?

File: test/api.js (hunk 2 above)
Comment (nottrobin) [inline test/api.js]: Yep good idea

File: lib/index.js (hunk 1 above)
Comment (nottrobin) [inline lib/index.js]: I've tidied this up.

File: test/api.js (hunk 2 above)
Comment (nottrobin) [inline test/api.js]: I started writing the code to save any initial value for SASS_PATH, then set it back to what it was afterwards. But then I thought it's actually quite unlikely that any changes to `process.env` would effect the environment outside of this particular `mocha` cycle, so I went to test it:

``` bash
$ export SASS_PATH=hello
$ mocha test/api.js
...
 129 passing (255ms)
$ echo $SASS_PATH
hello
```

So I think all I need to do it make sure SASS_PATH changes don't bleed through from one test to the other. So I've reset SASS_PATH to be an empty string before each test.

File: test/fixtures/include-path/lib-alternate/vars.scss (hunk 3 above)
Comment (nottrobin) [inline test/fixtures/include-path/lib-alternate/vars.scss]: I've tried to make my fixtures simpler and clearer. I hope it's the sort of thing you had in mind.

**Comment by nottrobin** [comment]
How can a travis build take quite so long? @nschonni is something up? Is there some way for me to start the travis build again?

**Comment by saper** [comment]
Looks like OSX jobs got stuck and couldn't even be started.

here's an update from @travisci  : https://www.traviscistatus.com/incidents/4mvp857qx8bw

**Comment by nottrobin** [comment]
Thanks @saper, I hadn't seen that. It's all passed now. @nschonni are you happy that I've address all your points?

**Comment by saper** [comment]
Looks good to me. Would that be possible to squeeze this into one commit?

**Comment by nottrobin** [comment]
@saper sure no problem

**Comment by xzyfer** [comment]
Nice work everyone. Added this to the next.minor milestone. I believe we have one other PR to land in 3.9.0 also.

**Comment by nottrobin** [comment]
Okay commits squashed into 3788c5d9570f2141ccbd6a574af21fcd57d63110.

**Comment by nottrobin** [comment]
That's odd, it's failing on a test which is nothing to do with this commit. Any ideas?

**Comment by saper** [comment]
Big thank you for the contribution!

**Comment by nottrobin** [comment]
No problem. Thanks for all your help with this @saper et al., it was fun! Do you know when the next minor release will be?
